stellargraph/mapper/padded_graph_generator.py

```diff
@@ -131,8 +132,13 @@ def flow(
                 f"expected batch_size to be strictly positive integer, found {batch_size}"
             )
 
+        if isinstance(graph_ilocs[0], StellarGraph):
```

Comment:
I think we need to do a lot more validation here; e.g. this allows a user to pass in an empty graph (#1338) or a graph without features or a heterogeneous graph.

One way to do this would be to factor out the loop (or just the loop body) from `__init__` and call it both there and here.

Given there's a variety of edge cases, I wonder if we should do the easy thing (emitting a better error) for 1.0 and do this more flexible version for 1.1?

Reply:
good point, I had a quick go at extending existing edge case tests to handle `.floaw` and I think they're all covered